rust/lance/src/dataset.rs

```diff
@@ -2590,6 +2630,143 @@ mod tests {
         assert!(matches!(result.unwrap_err(), Error::DatasetNotFound { .. }));
     }
 
+    #[tokio::test]
+    async fn test_drop() {
```

Comment:
Could we also assert that this operation preserves schema-level metadata?

Reply:
Added a check.